velox/functions/sparksql/specialforms/DecimalRound.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/expression/FunctionCallToSpecialForm.h"
+
+namespace facebook::velox::functions::sparksql {
+class DecimalRoundCallToSpecialForm : public exec::FunctionCallToSpecialForm {
+ public:
+  TypePtr resolveType(const std::vector<TypePtr>& argTypes) override;
+
+  exec::ExprPtr constructSpecialForm(
+      const TypePtr& type,
+      std::vector<exec::ExprPtr>&& compiledChildren,
+      bool trackCpuUsage,
+      const core::QueryConfig& config) override;
+
+  static constexpr const char* kRoundDecimal = "decimal_round";
+};
+
+/// Returns the result precision and scale after rounding from the input
+/// precision and scale to a new scale.
+std::pair<uint8_t, uint8_t>
+getResultPrecisionScale(uint8_t precision, uint8_t scale, int32_t roundScale);
```

Comment:
Should the type of roundScale match the type of scale?

Maybe make this function a static method in DecimalRoundCallToSpecialForm.

Reply:
RoundScale could be negative while scale cannot because it is from the input decimal.  In Spark, roundScale is integer, and its abs value could be greater than 38, so below cases are valid.
```
scala> spark.sql("select round(cast (12.3 as DECIMAL(3, 1)), 999)").show()

+--------------------------------------+                                       

|round(CAST(12.3 AS DECIMAL(3,1)), 999)|

+--------------------------------------+

|                                  12.3|

+--------------------------------------+

scala> spark.sql("select round(cast (12.3 as DECIMAL(3, 1)), -999)").show()

+---------------------------------------+                                      

|round(CAST(12.3 AS DECIMAL(3,1)), -999)|

+---------------------------------------+

|                                      0|

+---------------------------------------+
```